### PR TITLE
Clean `cuda.empty_cache` usage

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

The `empty_cache` function already checks if `is_initialized`:

https://github.com/pytorch/pytorch/blob/v1.4.1/torch/cuda/memory.py#L34-L35

And we don't need to set the root device as the default device has been set previously.

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [n/a] Did you make sure to update the documentation with your changes? (if necessary)
- [n/a] Did you write any new necessary tests? (not for typos and docs)
- [n/a] Did you verify new and existing tests pass locally with your changes?
- [n/a] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

### CODE REVIEW

File: pytorch_lightning/plugins/training_type/ddp_spawn.py [hunk 1]
```diff
@@ -310,7 +316,7 @@ def __recover_child_process_weights(self, best_path, last_path):
             self.lightning_module.load_state_dict(ckpt)
 
     def barrier(self, *args, **kwargs) -> None:
-        if not torch_distrib.is_initialized():
+        if not distributed_available():
```
Comment (awaelchli) [inline pytorch_lightning/plugins/training_type/ddp_spawn.py:319]: hey, it needs both, available() and initialized()

my bad in the barrier PR #8165, I thought available() is redundant but didn't notice the windows test failing who need this.

Comment (carmocca) [reply pytorch_lightning/plugins/training_type/ddp_spawn.py:319]: `distributed_available` has that check

Comment (awaelchli) [reply pytorch_lightning/plugins/training_type/ddp_spawn.py:319]: I would avoid that because it has an or with `tpu_distributed()` which uses a different library for distributed training and is not compatible with this ddp plugin, but ok not a big deal.

**Review by awaelchli — APPROVED** [review]
fingers crossed ;)

**Review by justusschock — APPROVED** [review]

**Review by ethanwharris — APPROVED** [review]

**Review by Borda — APPROVED** [review]
just curious what was the reason we did it before only for root?

**Comment by carmocca** [comment]
> just curious what was the reason we did it before only for root?

We set the default device:

https://github.com/PyTorchLightning/pytorch-lightning/blob/74eb6cc7e90b4b06c0136504101d6f5c343e93dc/pytorch_lightning/plugins/training_type/ddp.py#L373

So the context manager was uneccessary
